core/src/main/scala/io/delta/tables/execution/DeltaTableOperations.scala

```diff
@@ -76,11 +80,21 @@ trait DeltaTableOperations extends AnalysisHelper { self: DeltaTable =>
   }
 
   protected def executeRestore(
-      deltaLog: DeltaLog,
-      version: Option[Long] = None,
-      timestamp: Option[String] = None): DataFrame = {
-    RestoreTableCommand(deltaLog, version, timestamp).run(sparkSession)
-    sparkSession.emptyDataFrame
+      table: DeltaTableV2,
+      versionAsOf: Option[Long],
+      timestampAsOf: Option[String]): DataFrame = {
+    val identifier = table.getTableIdentifierIfExists.map(
+      id => Identifier.of(id.database.toArray, id.table))
+    val sourceRelation = DataSourceV2Relation.create(table, None, identifier)
```

Comment:
so basically instead of creating the command directly.. you are creating an intermediate logical plan... and then making DeltaAnalysis produce the DeltaCommand. 

I think i like it. this means that both SQL and Scala API will go through the same code paths.

Reply:
I think it will be good to highlight this change in the PR description. this is an important change.

Reply:
Sounds good, updated the PR description.